tests/accessor/accessor_api_common_buffer_local.h

```diff
@@ -259,34 +259,41 @@ struct buffer_accessor_get_pointer_w {
 /**
  *  @brief Tests buffer accessors pointer value with read-write data access
  */
-template <typename T, int dim, sycl::target target>
+template <typename T, int dim, sycl::access_mode mode, sycl::target target>
 struct buffer_accessor_get_pointer_rw {
   using error_code_t = buffer_accessor_api_pointer_error_code;
   using expected_value_t = buffer_accessor_expected_value<T, dim>;
 
-  template <typename accT, typename errorAccT>
-  static void check(const accT& acc, const errorAccT& errorAccessor,
+  template <sycl::access::placeholder placeholder>
+  using acc_t = sycl::accessor<T, dim, mode, target, placeholder>;
+
+  template <sycl::access::placeholder placeholder, typename errorAccT>
+  static void check(const acc_t<placeholder>& acc,
+                    const errorAccT& errorAccessor,
                     const sycl_id_t<dim>& offset) {
     const auto expectedRead = expected_value_t::expected_read(offset);
     const auto expectedWrite = expected_value_t::expected_write(offset);
+    constexpr bool noInitExpected =
+        (target == sycl::target::local) ||
+        (mode == sycl::access_mode::discard_read_write);
```

Comment:
Shouldn't we also check for `sycl::access_mode::discard_write`?

Reply:
`sycl::access_mode::discard_write` is for the write-only access, with no read access available. It is covered by different code path: `buffer_accessor_get_pointer_r`